### PR TITLE
EPMWL-7 Implement Executor

### PR DESCRIPTION
**Pull request by kshavonina**

### CODE REVIEW

File: src/main/java/com/epam/wl/executor/Executor.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+public class Executor {
+    private final Connection connection;
+
+    public Executor(Connection connection) {
+        this.connection = connection;
+    }
+
+    public void execUpdate(String update) throws SQLException {
+        Statement stmt = connection.createStatement();
```
Comment (TimWinnicheck) [inline src/main/java/com/epam/wl/executor/Executor.java]: Statement can be final

File: src/main/java/com/epam/wl/executor/Executor.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    public <T> T execQuery(String query, ResultHandler<T> handler) throws SQLException {
+        Statement stmt = connection.createStatement();
+        stmt.execute(query);
+        ResultSet result = stmt.getResultSet();
+        T value = handler.handle(result);
+        result.close();
+        stmt.close();
+
```
Comment (TimWinnicheck) [inline src/main/java/com/epam/wl/executor/Executor.java]: Statement and Resultset can be final

File: src/main/java/com/epam/wl/executor/Executor.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        return value;
+    }
+
+}
```
Comment (TimWinnicheck) [inline src/main/java/com/epam/wl/executor/Executor.java:32]: There's no reason to shorten the names, to make methods more readable I suggest renaming execUpdate and execQuery to executeUpdate and executeQuery

File: src/main/java/com/epam/wl/executor/Executor.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        this.connection = connection;
+    }
+
+    public void execUpdate(String update) throws SQLException {
+        Statement stmt = connection.createStatement();
+        stmt.execute(update);
+        stmt.close();
+    }
+
+    public <T> T execQuery(String query, ResultHandler<T> handler) throws SQLException {
```
Comment (ginsenghillock) [inline src/main/java/com/epam/wl/executor/Executor.java]: input parameters should be final

File: src/main/java/com/epam/wl/executor/Executor.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import java.sql.Statement;
+
+public class Executor {
+    private final Connection connection;
+
+    public Executor(Connection connection) {
+        this.connection = connection;
+    }
+
+    public void execUpdate(String update) throws SQLException {
```
Comment (ginsenghillock) [inline src/main/java/com/epam/wl/executor/Executor.java]: update variable should be final

File: src/main/java/com/epam/wl/executor/ResultHandler.java [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package com.epam.wl.executor;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public interface ResultHandler<T> {
+    T handle(ResultSet resultSet) throws SQLException;
```
Comment (ginsenghillock) [inline src/main/java/com/epam/wl/executor/ResultHandler.java]: resultSet variable should be final

**Comment by ginsenghillock** [comment]
Reviewed

**Pull request merged**